detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryAbstractClassSpec.kt

```diff
@@ -38,28 +33,56 @@ class UnnecessaryAbstractClassSpec : Spek({
                     }
                 """
                 val findings = subject.compileAndLintWithContext(env, code)
-                assertFindingMessage(
-                    findings,
-                    "An abstract class without a concrete member can be refactored to an interface."
-                )
+                assertFindingMessage(findings, message)
             }
 
-            it("reports completely-empty abstract classes") {
-                val code = """
-                    abstract class A
-                    abstract class B()
-                """
-                assertThat(subject.compileAndLintWithContext(env, code)).hasSize(2)
-            }
+            context("reports completely-empty abstract classes") {
+                it("case 1") {
+                    val code = "abstract class A"
+                    val findings = subject.compileAndLintWithContext(env, code)
+                    assertFindingMessage(findings, message)
+                }
 
-            it("does not report a completely-empty abstract class that inherits from an interface") {
-                val code = """
-                    interface A {
-                        val i: Int
-                    }
-                    abstract class B : A
-                """.trimIndent()
-                assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
+                it("case 2") {
+                    val code = "abstract class A()"
+                    val findings = subject.compileAndLintWithContext(env, code)
+                    assertFindingMessage(findings, message)
+                }
+
+                it("case 3") {
+                    val code = "abstract class A {}"
+                    val findings = subject.compileAndLintWithContext(env, code)
+                    assertFindingMessage(findings, message)
+                }
+
+                it("case 4") {
+                    val code = "abstract class A() {}"
+                    val findings = subject.compileAndLintWithContext(env, code)
+                    assertFindingMessage(findings, message)
+                }
+
+                it("that inherits from an interface") {
+                    val code = """
+                        interface A {
+                            val i: Int
+                        }
+                        abstract class B : A
+                    """.trimIndent()
```

Comment:
I don't think we used the trim statement for the code snippets except for asserts regarding the reported line number.

```suggestion
                    """
```